tests/bluetooth/bsim/audio/src/csip_set_coordinator_test.c

```diff
@@ -103,8 +103,7 @@ static void connected(struct bt_conn *conn, uint8_t err)
 	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
 
 	if (err != 0) {
-		bt_conn_unref(default_conn);
-		default_conn = NULL;
+		bt_conn_unref(conn);
```

Comment:
Nice catch. I had left CSIP alone since the Set Member test needs special handling, but it's fine to use the flag for the set coordinator. 

> slightly related: do we have control over which of the callback functions will be used, is it always the last one being registered?

Both callbacks will be called; it is a linked list :)